src/rust/engine/src/nodes.rs

```diff
@@ -901,10 +904,16 @@ impl Snapshot {
           // And then create a Snapshot.
           stats
```

Comment:
The `Stat` type (which needed to fake the file access) should be completely removed in favor of `FileDigest` (which is actually usefully memoized): just make sure that you update this codepath https://github.com/pantsbuild/pants/blob/9ebddaf7b9255e03bd9d3afa6744c284098ae4c3/src/rust/engine/src/nodes.rs#L1154-L1165 to link `FileDigest` to its canonical stat.

Reply:
Done. Modified fs_subject to avoid relying on your review to do this in the future.